zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryRequestMapper.java

```diff
@@ -104,32 +113,22 @@ public static Either<ProblemDetail, SearchQueryPage> toSearchQueryPage(
     return Either.right(null);
   }
 
-  public static Either<ProblemDetail, ProcessInstanceSort> toSearchQuerySort(
-      final List<SearchQuerySortRequest> sorting) {
+  private static <T, B extends AbstractBuilder<B> & ObjectBuilder<T>>
+      Either<ProblemDetail, T> toSearchQuerySort(
+          final List<SearchQuerySortRequest> sorting,
+          final Supplier<B> builderSupplier,
+          BiConsumer<String, B> sortFieldConsumer) {
     if (sorting != null && !sorting.isEmpty()) {
-      final var builder = SortOptionBuilders.processInstance();
-
+      final var builder = builderSupplier.get();
       for (SearchQuerySortRequest sort : sorting) {
-        final var field = sort.getField();
-        final var order = sort.getOrder();
-
-        if ("processInstanceKey".equals(field)) {
-          builder.processInstanceKey();
-        } else if ("startDate".equals(field)) {
-          builder.startDate();
-        } else if ("endDate".equals(field)) {
-          builder.endDate();
-        } else {
-          throw new RuntimeException("unkown sortBy " + field);
+        if (sort.getField() == null) {
```

Comment:
🔧 If we add such validation constraints, we should add unit tests in the REST gateway as well. Since we don't have any tests for the `ProcessInstanceController` yet, now might be a good time to add some.

Reply:
Agree, let me add these tests

Reply:
I created `ProcessInstanceControllerTest`